### PR TITLE
rewrite promisification of formidable form

### PR DESCRIPTION
**Pull request by B0ER**
https://github.com/dlau/koa-body/issues/191

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -172,40 +172,21 @@ function isMultiPart(ctx, opts) {
...
-    var fields = {};
-    var files = {};
     var form = new forms.IncomingForm(opts);
-    form.on('end', function () {
+
+    if (opts.onFileBegin) {
+      form.on('fileBegin', opts.onFileBegin);
+    }
+
+    form.parse(ctx.req, (err, fields, files) => {
```
Comment (MarkHerhold) [inline index.js:181]: wow that's clean!

**Comment by MarkHerhold** [comment]
Sorry for the delay. Life is busy. At first glance, I like this. Let me review it further and refresh my memory of what tests there are to make sure all our bases are covered.

**Comment by MarkHerhold** [comment]
I do think we'll need to do a major version bump if we change the default. Is there a significant reason to change it?

**Comment by B0ER** [comment]
No, there is no need to change it. Koa-body works correct. I've found an issue with the formidable library. It returns a file-object with the key "photos[]", while it has to return an array with this file-object (an array of one element). Also, formidable.opts.multiple = true.

**Pull request closed without merging**